src/app/shared/components/common/in-place-edit/in-place-edit.component.ts

```diff
@@ -1,88 +1,42 @@
-import { DOCUMENT } from '@angular/common';
-import {
-  AfterViewInit,
-  ChangeDetectionStrategy,
-  ChangeDetectorRef,
-  Component,
-  ElementRef,
-  EventEmitter,
-  Inject,
-  OnDestroy,
-  Output,
-} from '@angular/core';
-import { Subject, fromEvent } from 'rxjs';
-import { filter, map, takeUntil } from 'rxjs/operators';
+import { ChangeDetectionStrategy, Component, ElementRef, EventEmitter, Output } from '@angular/core';
 
 @Component({
   selector: 'ish-in-place-edit',
   templateUrl: './in-place-edit.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush,
   styleUrls: ['./in-place-edit.component.scss'],
 })
-export class InPlaceEditComponent implements AfterViewInit, OnDestroy {
+export class InPlaceEditComponent {
   @Output() edited = new EventEmitter<void>();
   @Output() aborted = new EventEmitter<void>();
 
   private mode: 'view' | 'edit' = 'view';
-  private destroy$ = new Subject<void>();
 
-  hover: boolean;
-
-  constructor(
-    private host: ElementRef,
-    private cdRef: ChangeDetectorRef,
-    @Inject(DOCUMENT) private document: Document
-  ) {}
-
-  ngAfterViewInit() {
-    fromEvent(this.document, 'mousedown')
-      .pipe(
-        // only main button clicks
-        filter((event: MouseEvent) => !event.button),
-        map(({ target }) => this.host.nativeElement.contains(target)),
-        takeUntil(this.destroy$)
-      )
-      .subscribe(click => {
-        const newMode = click ? 'edit' : 'view';
-        if (newMode === 'view' && newMode !== this.mode) {
-          this.confirm();
-        }
-        this.mode = newMode;
-        this.cdRef.detectChanges();
-        if (this.mode === 'edit') {
-          setTimeout(() => {
-            this.host.nativeElement.querySelector('.form-control')?.focus();
-          }, 200);
-        }
-      });
+  constructor(private host: ElementRef) {}
+
+  changeEditMode() {
+    if (this.mode === 'edit') {
+      setTimeout(() => {
+        this.host.nativeElement.querySelector('.form-control')?.focus();
+      }, 200);
+    }
+    if (this.mode === 'view') {
+      this.confirm();
+      this.mode = 'edit';
+    }
```

Comment:
has better functionality for keyboard, but the text itself isn't clickable anymore to enter the form-field, that functionality has to be restored

Reply:
Has been reverted to edit by clicking the text, but also preserving the button functionality of the edit-pen